training/01_introduction.Rmd

```diff
@@ -48,6 +48,21 @@ This training assumes basic knowledge of R and RStudio. If you want a quick R re
 
 Throughout this training we will occasionally use the namespace syntax `package_name::function_name()` when writing a function. This syntax denotes which package a function came from.  For example `dataone::getSystemMetadata` selects the `getSystemMetadata` function from the `dataone` R package. More detailed information on namespaces can be found [here](http://r-pkgs.had.co.nz/namespace.html).
 
+## A note on effective troubleshooting in R
+We suggest using a combination of **m**inimal **r**eproducible **e**xamples (MRE) and the package `reprex` to create **rep**roducible **ex**amples. This will allow others to better help you if we can run the code on our own computers.
+
+A MRE is stripping down your code to only the parts that cause the bug.
+
+How to generate a reprex:
+
+1. copy the code you want to ask about
+2. call `reprex()` 
+3. fix until everything runs smoothly
+4. copy the result to ask your question
+
+For more information and examples check out more of Jenny Bryan's [slides](https://speakerdeck.com/jennybc/reprex-help-me-help-you)
+or watch the [video](https://vimeo.com/208749032) starting at about the 10 min mark.
+
```

Comment:
You might want to add a note here that any publishing function will not work with reprex. However, the same principals of generating a minimum reproducible example can still be followed. When we are generating our MRE's, try to determine which step is having the issue. Is it the EML editing or the publishing, for example? 

Here is a good MRE for an EML issue that I had which might be helpful to include.

```
me <- list(individualName = list(givenName = "Jeanette", surName = "Clark"))

attributes <- data.frame(attributeName = 'length_1',
                         attributeDefinition = 'def1',
                         measurementScale = 'ratio',
                         domain = 'numericDomain',
                         unit = 'meter',
                         numberType = 'real',
                         stringsAsFactors = FALSE)

att_list <- set_attributes(attributes)


doc_ex <- list(packageId = "id", system = "system", 
            dataset = list(title = "A Mimimal Valid EML Dataset",
                           creator = me,
                           contact = me,
                           dataTable = list(entityName = "data table", attributeList = att_list))
)

get_attributes(doc_ex$dataset$dataTable)
```